lib/src/widgets/chat_groupedlist_widget.dart

```diff
@@ -287,82 +286,142 @@ class _ChatGroupedListWidgetState extends State<ChatGroupedListWidget>
+                  final message = messages[newIndex];
+                  return ChatBubbleWidget(
+                    key: message.key,
+                    messageTimeTextStyle:
+                        chatBackgroundConfig.messageTimeTextStyle,
+                    messageTimeIconColor:
+                        chatBackgroundConfig.messageTimeIconColor,
+                    message: message,
+                    messageConfig: widget.messageConfig,
+                    chatBubbleConfig: chatBubbleConfig,
+                    profileCircleConfig: profileCircleConfig,
+                    swipeToReplyConfig: widget.swipeToReplyConfig,
+                    repliedMessageConfig: widget.repliedMessageConfig,
+                    slideAnimation: _slideAnimation,
+                    onLongPress: (yCoordinate, xCoordinate) =>
+                        widget.onChatBubbleLongPress(
+                      yCoordinate,
+                      xCoordinate,
+                      message,
+                    ),
+                    onSwipe: widget.assignReplyMessage,
+                    shouldHighlight: state == message.id,
+                    onReplyTap: widget
+                                .repliedMessageConfig
+                                ?.repliedMsgAutoScrollConfig
+                                .enableScrollToRepliedMsg ??
+                            false
+                        ? (replyId) => _onReplyTap(replyId, snapshot.data)
+                        : null,
                   );
                 },
-              )
-            : Center(
-                child: chatBackgroundConfig.loadingWidget ??
-                    const CircularProgressIndicator(),
               );
+            },
+          );
+        } else {
+          return Center(
+            child: chatBackgroundConfig.loadingWidget ??
+                const CircularProgressIndicator(),
+          );
+        }
       },
     );
   }
+
+  List<Message> sortMessage(List<Message> messages) {
+    final elements = [...messages];
+
+    elements.sort(
+      widget.chatBackgroundConfig.messageSorter ??
+          (a, b) => a.createdAt.compareTo(b.createdAt),
+    );
+    if (widget.chatBackgroundConfig.groupedListOrder.isAsc) {
+      return elements;
+    } else {
+      return elements.reversed.toList();
+    }
+  }
+
+  /// return DateTime by checking lastMatchedDate and message created DateTime
+  DateTime _groupBy(
+    Message message,
+    DateTime lastMatchedDate,
+  ) {
+    /// If the conversation is ongoing on the same date,
+    /// return the same date [lastMatchedDate].
+
+    /// When the conversation starts on a new date,
+    /// we are returning new date [message.createdAt].
+    return lastMatchedDate.getDateFromDateTime ==
+            message.createdAt.getDateFromDateTime
+        ? lastMatchedDate
+        : message.createdAt;
+  }
+
+  Widget _groupSeparator(DateTime createdAt) {
+    return featureActiveConfig?.enableChatSeparator ?? false
```

Comment:
Try avoiding performing other calculations for chat separator if it is disabled.